xcell/cls/cl.py

```diff
@@ -173,24 +180,35 @@ def get_cl_file(self):
                 nl_cp = mapper1.get_nl_coupled()
                 nl = w.decouple_cell(nl_cp)
             else:
-                nmap1 = 1 + (mapper1.get_spin() > 0)
-                nmap2 = 1 + (mapper2.get_spin() > 0)
-                nl_cp = np.zeros((nmap1 * nmap2, 3 * self.nside))
-                nl = np.zeros([nmap1 * nmap2, self.b.get_n_bands()])
+                n_cls = self.get_n_cls()
+                nl_cp = np.zeros((n_cls, 3 * self.nside))
+                nl = np.zeros([n_cls, self.b.get_n_bands()])
             # Signal
             if auto and mapper1.custom_auto:
                 cl_cp = mapper1.get_cl_coupled()
                 cl = w.decouple_cell(cl_cp)
+                cls_cov = mapper1.get_cls_covar_coupled()
+                cl_cov_cp = cls_cov['cross']
+                cl_cov_11_cp = cls_cov['auto_11']
+                cl_cov_12_cp = cls_cov['auto_12']
+                cl_cov_22_cp = cls_cov['auto_22']
                 # mapper.get_cl_coupled is assumed to return
                 # the noise-less power spectrum. No need
                 # to subtract it here.
             else:
-                cl_cp = nmt.compute_coupled_cell(f1, f2)
-                cl = w.decouple_cell(cl_cp)
-                cl_cp -= nl_cp
+                cl_cov_cp = nmt.compute_coupled_cell(f1, f2)
+                cl_cov_11_cp = cl_cov_cp
+                cl_cov_12_cp = cl_cov_cp
+                cl_cov_22_cp = cl_cov_cp
+                cl = w.decouple_cell(cl_cov_cp)
+                cl_cp = cl_cov_cp - nl_cp
                 cl -= nl
             np.savez(fname, ell=ell, cl=cl, cl_cp=cl_cp, nl=nl,
-                     nl_cp=nl_cp, wins=wins)
+                     nl_cp=nl_cp, cl_cov_cp=cl_cov_cp,
```

Comment:
We should comment somewhere that `cl_cov_X` includes the noise contribution

Reply:
Done